TeamCode/src/main/java/org/firstinspires/ftc/teamcode/api/hw/SmartServo.java

```diff
@@ -99,25 +99,24 @@ private void scanServo(double position, int ms) {
     }
 
     public void scanServoAsync(double position, int ms) {
-        if (lock) {
+        if (locked) {
             return;
         }
-        lock = true;
-        OldRobot.executor.execute(new Runnable() {
-            @Override
-            public void run() {
-                scanServo(position, ms);
-            }
-        });
-        lock = false;
+        locked = true;
+        Robot.executorService.execute(() -> scanServo(position, ms));
+        locked = false;
     }
 
     public void scanServoSync(double position, int ms) {
```

Comment:
Let's add another function that "steps" the servo with a time interval (ie step the servo, note down the last used time. If the last used time is far enough, then step again. Increasing the time interval can make movements slower, and thus can be controlled from teleop in a semi-async method in that it doesn't block the main thread but we can enforce sleeps on it.

Reply:
I'm unsure how to do this, you may have to